File: C++/프로그래머스/0/120883. 로그인 성공？/로그인 성공？.cpp
#include <string>
#include <vector>

using namespace std;

const string SUCCESS = "login";
const string FAILED = "fail";
const string WRONGPW = "wrong pw";

string solution(vector<string> id_pw, vector<vector<string>> db) {
    // db 순회
    for (const auto& e: db) {
        // id 찾음
        if (e[0] == id_pw[0]) {
            // pw 비교
            if (e[1] == id_pw[1]) return SUCCESS;
            return WRONGPW;
        }
    }
    return FAILED;
}

File: Go/프로그래머스/0/120883. 로그인 성공？/로그인 성공？.go
func solution(id_pw []string, db [][]string) string {
    for _, account := range db {
        if id_pw[0] == account[0] {
            if id_pw[1] == account[1] {
                return "login"
            }
            return "wrong pw"
        }
    }
    return "fail"
}